ethereumj-core/src/main/java/org/ethereum/facade/Ethereum.java

```diff
@@ -53,8 +58,37 @@
 
     Blockchain getBlockchain();
 
+    /**
+     * Uses expensive enough logic for delivery events to client.<br>
+     * Will be removed in future releases.
+     *
+     * @param listener
+     * @deprecated use {@link #subscribe(Subscription)} instead.
+     */
+    @Deprecated
     void addListener(EthereumListener listener);
 
+    /**
+     * Adds subscription for specific event.
+     *
+     * @param subscription
+     * @return publisher instance to support fluent API.
+     * @see Subscription
+     * @see Publisher
+     * @see Event
+     */
+    Publisher subscribe(Subscription subscription);
+
+    /**
+     * More convenient version of {@link #subscribe(Subscription)}
+     */
+    <T> Publisher subscribe(Class<? extends Event<T>> type, Consumer<T> handler);
+
+    /**
+     * More convenient version of {@link #subscribe(Subscription)}
+     */
+    <T> Publisher subscribe(Class<? extends Event<T>> type, BiConsumer<T, Subscription.LifeCycle> handler);
```

Comment:
I think `Publisher` class should also have such shortcuts

Reply:
To increase UX we may create shortcuts for event classes and put them together into one place.
For example, use `Event` or create new `EventType` class and add a list of event types definition with readable names:
`public static final Class<? extends Event<BlockAdded.Data>> BLOCK_ADDED = BlockAdded.class;`
Then point this type aggregation class in javadocs. It would work like an event picker then and make user choice faster, user will have to type `Event.` to get a list of available options